cypress/e2e/testing/loginlogout.cy.js

```diff
@@ -0,0 +1,64 @@
+const validEmail = "MyNameAgain24@noroff.no";
+const validPassword = "mynameagain24";
+
+const invalidEmail = "invalidEmail@noroff.no";
+const invalidPassword = "invalidPassword";
+
+describe("Login and logout test", () => {
+  beforeEach(() => {
+    cy.visit("/");
+    cy.clearLocalStorage();
+    cy.wait(1000);
+  });
+
+  it("Login with valid credentials", () => {
+    cy.get("#registerForm button[data-auth='login']").click();
+    cy.wait(1000);
+    cy.get("#loginEmail").type(validEmail);
+    cy.get("#loginPassword").type(validPassword);
+    cy.get("#loginForm button[type='submit']").click();
+    cy.wait(1000);
+    cy.get("button[data-visible='loggedIn']").should("exist");
+  });
+
+  it("Logging out using the logout button", () => {
+    cy.get("#registerForm button[data-auth='login']").click();
+    cy.wait(1000);
+    cy.get("#loginEmail").type(validEmail);
+    cy.get("#loginPassword").type(validPassword);
+    cy.get("#loginForm button[type='submit']").click();
+    cy.wait(1000);
+    cy.get("button[data-visible='loggedIn']").should("exist");
+    cy.get("button[data-visible='loggedIn']").should("be.visible");
+    cy.wait(1000);
+    cy.get("button[data-visible='loggedIn']").click();
+    cy.wait(1000);
+    cy.get("button[data-visible='loggedOut']").should("exist");
+    cy.get("button[data-visible='loggedOut']").should("be.visible");
+  });
+
+  it("Try to login with invalid credentials, but fails and shows a message", () => {
+    cy.intercept(
+      "POST",
+      "https://nf-api.onrender.com/api/v1/social/auth/login",
+    ).as("loginAttempt");
+
+    cy.get("#registerForm button[data-auth='login']").click();
+    cy.wait(1000);
+    cy.get("#loginEmail").type(invalidEmail);
+    cy.get("#loginPassword").type(invalidPassword);
+    cy.on("window:alert", (str) => {
+      expect(str).to.equal(
+        "Either your username was not found or your password is incorrect",
+      );
+    });
+    cy.get("#loginForm button[type='submit']").click();
+
+    cy.wait("@loginAttempt").then((interception) => {
+      assert.equal(interception.response.statusCode, 401);
+    });
+
+    cy.wait(1000);
+    cy.get("button[data-visible='loggedOut']").should("be.visible");
+  });
+});
```

Comment:
(describe) An Apple (it) is red
(describe) A Camel (it) has humps
(describe) createPost (it) shows an error when you send an empty body

Reply:
Nothing just trying to show how to improve the test names, you can resolve

Reply:
Ah ok, thanks I fixed it :)